### PR TITLE
feature(attribute) add option sorting endpoint

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Add sorting option endpoint ```PUT /attributes/{attribute}/options/{option}/move```

Issue #1685
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wiewiurdp — DISMISSED** [review]

File: module/attribute/features/attribute-multi-select.feature [hunk 1]
```diff
@@ -140,7 +140,7 @@ Feature: Multi multi select attribute manipulation
     And the JSON node "label.en_GB" should contain "Option en 1 updated"
     And the JSON node "code" should contain "option_1_updated"
 
-  Scenario: Get attribute options
+  Scenario: Scenario: Check attribute options sort after adding all
```
Comment (piotrkreft) [inline module/attribute/features/attribute-multi-select.feature:143]:
```suggestion
 Scenario: Check attribute options sort after adding all
```

File: module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php [hunk 2]
```diff
@@ -9,28 +9,100 @@
...
+                $this->connection->insert(
+                    self::TABLE,
+                    [
+                        'attribute_id' => $event->getAggregateId()->getValue(),
+                        'option_id' => $event->getOptionId()->getValue(),
+                        'index' => $to,
+                    ]
+                );
+            } else {
+                var_dump($to);
```
Comment (piotrkreft) [inline module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php:49]:
```suggestion
```

File: module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php [hunk 3]
```diff
@@ -9,28 +9,100 @@
...
+                );
+
+
+
+                $this->mergePosition($event->getAggregateId(), $from);
+
+                //  $this->print($event);
+            }
+        } catch (\Exception $exception) {
+            var_dump($exception->getMessage());
```
Comment (piotrkreft) [inline module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php:79]: Probably lots to check still... ;)

File: module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php [hunk 4]
```diff
@@ -9,28 +9,100 @@
...
+            ->setParameter(':id', $event->getAggregateId()->getValue())
+            ->orderBy('index')
+            ->execute()
+            ->fetchAllAssociative();
+
+        foreach ($result as $line) {
+            echo $line['index'].' : '.$line['option_id'].PHP_EOL;
+        }
+
+        var_dump('------------------------');
```
Comment (piotrkreft) [inline module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php:106]: and here

File: module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php [hunk 5]
```diff
@@ -9,28 +9,100 @@
...
-                'attribute_id' => $event->getAggregateId()->getValue(),
-                'option_id' => $event->getOptionId()->getValue(),
-                'index' => $option,
-            ]
-        );
+        try {
+            $from = $this->getPosition($event->getAggregateId(), $event->getOptionId());
+            $to = $event->getIndex();
+
+            var_dump('from '.$from);
```
Comment (piotrkreft) [inline module/attribute/src/Infrastructure/Persistence/Projector/Attribute/DbalAttributeOptionMovedEventProjector.php:22]: here

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Pull request merged**